integration/backward_compatibility_test.go

```diff
@@ -17,69 +17,71 @@ import (
 )
 
 const (
+	networkName = "e2e-cortex-test"
 	// If you change the image tag, remember to update it in the preloading done
 	// by CircleCI too (see .circleci/config.yml).
 	previousVersionImage = "quay.io/cortexproject/cortex:v0.6.0"
 )
 
 func TestBackwardCompatibilityWithChunksStorage(t *testing.T) {
-	s, err := e2e.NewScenario()
+	s, err := e2e.NewScenario(networkName)
 	require.NoError(t, err)
 	defer s.Close()
 
 	// Start dependencies.
-	require.NoError(t, s.StartService(e2edb.NewDynamoDB()))
-	require.NoError(t, s.StartService(e2edb.NewConsul()))
-	require.NoError(t, s.WaitReady("consul", "dynamodb"))
+	dynamo := e2edb.NewDynamoDB()
+	consul := e2edb.NewConsul()
+	require.NoError(t, s.StartAndWaitReady(dynamo, consul))
 
 	// Start Cortex components (ingester running on previous version).
 	require.NoError(t, ioutil.WriteFile(
 		filepath.Join(s.SharedDir(), cortexSchemaConfigFile),
 		[]byte(cortexSchemaConfigYaml),
 		os.ModePerm),
 	)
-	require.NoError(t, s.StartService(e2ecortex.NewTableManager("table-manager", ChunksStorage, "")))
-	require.NoError(t, s.StartService(e2ecortex.NewIngester("ingester-1", ChunksStorage, previousVersionImage)))
-	require.NoError(t, s.StartService(e2ecortex.NewDistributor("distributor", ChunksStorage, "")))
-	require.NoError(t, s.WaitReady("distributor", "ingester-1", "table-manager"))
+	tableManager := e2ecortex.NewTableManager("table-manager", ChunksStorage, previousVersionImage)
+	ingester1 := e2ecortex.NewIngester("ingester-1", consul.NetworkHTTPEndpoint(networkName), ChunksStorage, "")
+	distributor := e2ecortex.NewDistributor("distributor", consul.NetworkHTTPEndpoint(networkName), ChunksStorage, "")
+	require.NoError(t, s.StartAndWaitReady(distributor, ingester1, tableManager))
 
 	// Wait until the first table-manager sync has completed, so that we're
 	// sure the tables have been created.
-	require.NoError(t, s.Service("table-manager").WaitMetric("cortex_dynamo_sync_tables_seconds", 1))
+	require.NoError(t, tableManager.WaitMetric("cortex_dynamo_sync_tables_seconds", 1))
 
-	// Wait until the distributor has updated the ring.
-	require.NoError(t, s.Service("distributor").WaitMetric("cortex_ring_tokens_total", 512))
+	// Wait until both the distributor and querier have updated the ring.
```

Comment:
`Wait until the distributor has updated the ring.` (we're not waiting for the querier here)